src/Test/Perf/ConsumptionParser/Program.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Roslyn.Test.Performance.Utilities.ConsumptionParser
+{
+    class Program
+    {
+        public static void Main()
+        {
+            const string prefix = "PerfResults-";
+            const string share = @"\\mlangfs1\public\basoundr\PerfTraces\";
+
+            var runs = new List<Tuple<string, ConsumptionParse, RunInfo>>();
+
+            // Collect run information from the share
+            foreach (var directory in Directory.GetDirectories(share))
+            {
+                var lastPath = directory.Substring(share.Length);
+                if (!lastPath.StartsWith(prefix))
+                {
+                    continue;
+                }
+
+                var date = lastPath.Substring(prefix.Length);
+                var consumptionXml = Path.Combine(directory, "ConsumptionTempResults.xml");
+                var resultJson = Directory.EnumerateFiles(directory, "Roslyn*.json").First();
+
+                var parse = ConsumptionParse.Parse(File.ReadAllText(consumptionXml));
+                var runInfo = RunInfo.Parse(File.ReadAllText(resultJson));
+                runs.Add(Tuple.Create(date, parse, runInfo));
+
+            }
+
+            // Sort by date
+            runs.Sort((run1, run2) => run1.Item1.CompareTo(run2.Item1));
+
+            // Collect all of the metric names first 
+            var metrics = new HashSet<string>();
+            metrics.UnionWith(
+                from run in runs
+                from scenario in run.Item2.Scenarios
+                from metric in scenario.Counters
+                select metric.Name
+            );
+
+            // Write the column titles
+            Console.Write("build, username, branch, scenario");
+            foreach (var metric in metrics)
+            {
+                Console.Write($", z_{metric}");
```

Comment:
So that in PowerBI all of the metrics are grouped together in the alphabetically sorted list of fields.

Dirty hack, but the PowerBI usability was really awful without it.